senseicore/src/channels.rs

```diff
@@ -87,23 +82,28 @@ impl ChannelOpener {
 
     pub async fn open_batch(
         &mut self,
-        requests: Vec<ChannelOpenRequest>,
-    ) -> Vec<(ChannelOpenRequest, Result<[u8; 32], Error>)> {
-        let mut requests_with_results = requests
+        requests: Vec<OpenChannelRequest>,
+    ) -> Vec<(OpenChannelRequest, Result<[u8; 32], Error>)> {
+        let requests = requests
             .into_iter()
-            .map(|request| {
-                let result = self.initiate_channel_open(&request);
-
-                (request, result)
+            .map(|request| OpenChannelRequest {
+                custom_id: Some(
+                    request
+                        .custom_id
+                        .unwrap_or_else(|| thread_rng().gen_range(1..u64::MAX)),
+                ),
+                ..request
             })
             .collect::<Vec<_>>();
+        let mut requests_with_results = vec![];
+        let mut filters = vec![];
 
-        let filters = requests_with_results
-            .iter()
-            .filter(|(_request, result)| result.is_ok())
-            .map(|(request, _result)| {
+        for request in requests {
+            let result = self.initiate_channel_open(&request).await;
+
+            if result.is_ok() {
```

Comment:
This seems like an anti-pattern. Why move away from calling `map`? Also generally it is better to `match` instead of calling `is_ok`

Reply:
oh I think I remember why I did this now.  In this PR I ended up having to making initiate_channel_open async function because it now handles connecting to peer if needed.  I think it's not easy to async map but it's trivial to do it in a for loop.

Reply:
Oh okay, did not know that!